### ISSUE TITLE
Broken screen on tmux with emoji in status bar

### ISSUE THREAD

**Issue by voanhduy1512** [jwilm/alacritty#975]
Which operating system does the issue occur on?
MacOs
If on linux, are you using X11 or Wayland?
Nah

If i run tmux which has an emoji (from https://github.com/Goles/Battery, run which `battery -pt`, result is `#[fg=green][100%] ⚡#[default]%`) in status bar, iTerm2 display it probably but alacritty seem broken.

<img width="1440" alt="screenshot 2017-12-29 14 11 36" src="https://user-images.githubusercontent.com/1771266/34431498-52c4ebc0-eca2-11e7-95fb-1ab3975690aa.png">

Left side is iterm2, right side is alacritty

**Comment by Ajnasz** [comment]
Same on Linux, X11.

**Comment by cryptoeraser** [comment]
Same issue on OSX with Powerline. Anything fancy in the tmux status bar breaks when running inside alacritty. I have disabled any fancy icons and plugins for now. Looking forward to hear on this issue if any solutions are available.

**Comment by sodiumjoe** [comment]
is this an issue with emoji in general?

Have you checked your `locale` settings? Should look something like this:
```bash
$ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

Are you starting your shell with `--login`?

**Comment by sodiumjoe** [comment]
I can't repro this with just the https://github.com/Goles/Battery example. I'm using that in my tmux statusline:

```tmux
set -g status-right '%-a %-b %-e %-l:%M %p #(/usr/local/bin/battery -t)'
```

**Comment by cryptoeraser** [comment]
Hi @sodiumjoe, I think this happens with powerline enabled. Here: [Poerline](https://github.com/powerline/powerline)

**Comment by sodiumjoe** [comment]
sorry, I don't use powerline and don't intend to. It's also kind of a huge surface area to try to figure out what's going on. If you can find a minimal repro case, I think that would be very helpful.

In any case, emoji in tmux statusline works fine for me. Can you try the steps I outlined in https://github.com/jwilm/alacritty/issues/975#issuecomment-378317878 and report back?

**Comment by Ajnasz** [comment]
In tmux if I execute the command:`set -g status-right 'before ⚡ after'`the _r_ of the _after_ word appears on the new line.
Also when I try to edit that command in the tmux command prompt, after every keystroke the previous state of the command drawn above  a prompt until. That behavior stops when I delete the ⚡ char.

![shot-2018-04-06-104624](https://user-images.githubusercontent.com/38329/38411694-ee3be2f2-3987-11e8-9f74-74ede9f6d102.png)

Here I opened tmux prompt by `C-a :`, then pressed `up` then the `left` a few times, then deleted two a space and the utf8 char.

My font is *Envy Code R*

My locale is:
```
$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US:
LC_CTYPE=en_US.UTF-8
LC_NUMERIC=hu_HU.UTF-8
LC_TIME=hu_HU.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=hu_HU.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=hu_HU.UTF-8
LC_NAME=hu_HU.UTF-8
LC_ADDRESS=hu_HU.UTF-8
LC_TELEPHONE=hu_HU.UTF-8
LC_MEASUREMENT=hu_HU.UTF-8
LC_IDENTIFICATION=hu_HU.UTF-8
```
However when I  set `LC_ALL` to `en_US.UTF-8`, it doesn't change the behavior.

> Are you starting your shell with --login?

I login through lightdm, so I guess it's not a login shell.

**Comment by Ajnasz** [comment]
Strange thing is that no issue with ❇ char, only with ⚡.

**Comment by cryptoeraser** [comment]
@Ajnasz Yes, same here. But this happens in tmux. I guess nothing to do with Alacritty. I have tried it at my Linux box at work on tmux 2.6. Exactly same thing happened, as you described it. So it must be a tmux issue?

I am not sure, but it is good to see this is finally getting some traction here. I will try to remove the lightning emoji from the powerline setup and see if it fixes things for tmux, and then possibly for Alacritty.

**Comment by Ajnasz** [comment]
@cryptoeraser I don't think it's tmux issue, because in other terminal emulator it works well and I think the issue I described in my previous comment is related.

**Comment by cryptoeraser** [comment]
@Ajnasz Something is broken with that `⚡`. It breaks things in all my terminals on Linux. Not even inside tmux.

**Comment by MithrilTuxedo** [comment]
I've encountered the behaviour @Ajnasz is describing without Alacritty. I had my $PS1 configured to use a different emoji on each invocation, but in tmux the line would sporadically shift one-character left visually. It showed locally (macOS & iTerm2) before I figured out how to make sure tmux knew it would be using UTF-8.

I haven't figure out how to make the same issue not show up in a tmux session on macOS with a panel open to a tmux session on an Ubuntu server (at home) with a panel open to a tmux session on a Mac Mini, but I that happens entirely within iTerm2 (or the terminal on Ubuntu when I ever connect a display).


I get the vertical shift in Alacritty, along with a vertical flickering. I'm using Roboto Mono for Powerline. The bottom line of the screen is a blank line with the same background as the rightmost element of the powerline bar, and when I type I'm overtyping the powerline status bar that was shifted up.

Cursor on the `0`.
<img width="1440" alt="screen shot 2018-04-18 at 2 00 03 pm" src="https://user-images.githubusercontent.com/2111300/38958157-4644fb2c-4311-11e8-8092-7bf904996b6c.png">

After executing the command. Tmux/Alacritty will flicker and the status bar will move back to the line I'm typing on, above the line it should be on.
<img width="1440" alt="screen shot 2018-04-18 at 2 00 17 pm" src="https://user-images.githubusercontent.com/2111300/38958159-494eb8c6-4311-11e8-9c78-c5a191386865.png">
